### ISSUE TITLE
 [Link list]: QA: Links trigger a modal dialog hence it can be given button role

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9867]
### Description

Page Element:
The video CTA links that open video modal

Issue Description:
Since these links trigger A modal dialog, these can be given button role.

Recommend Fix:
As an accessibility best practice, Since this link performs the action of triggering a modal dialog, this can be given button role.

Test Tool:
JAWS FF
Voiceover safari iPhone

### Component(s) impacted

Link list - CTA type video ( web component canary and web component staging)

### Browser

Safari, Firefox

### Carbon for IBM.com version

v1.40.0-rc.0

### Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

Web component Canary https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components Web component staging https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
